composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -141,182 +168,147 @@ class AlgoliaSearcher extends React.Component {
 	}
 
 	/**
-	 * Sets all values required to display the detail view of a search result.
+	 * Adds the data from the clicked result to the information associated with the current search string.
 	 *
-	 * @param {number} resultIndex The index of the article you want to
-	 *                                  show in the state.results array.
-	 * @returns {void}
+	 * @param {number} currentDetailViewIndex The current detail view index.
+	 *
+	 * @returns {Object} Object containing the currently used queries with additional post data.
 	 */
-	showDetail( resultIndex ) {
+	addResultToUsedQueries( currentDetailViewIndex ) {
+		let post = this.state.results[ currentDetailViewIndex ];
 		let usedQueries = this.state.usedQueries;
-		let post = this.state.results[ resultIndex ];
 
 		usedQueries[ this.state.searchString ][ post.objectID ] = {
-			title: post.post_title,
+			title: post.postTitle,
```

Comment:
ESLint disagrees but this should really be `post_title`. The returned  `post` object doesn't have a `postTitle ` property, it does have a `post_title ` one. Maybe disable the ESLint rule for this line?

Note, the post title saved in the used queries is actually `undefined` because of this:

<img width="734" alt="screen shot 2017-09-18 at 14 45 36" src="https://user-images.githubusercontent.com/1682452/30542838-1e0b297e-9c81-11e7-9af9-6e34c0b6e30e.png">